include/realtime_tools/realtime_publisher.hpp

```diff
@@ -196,37 +177,23 @@ class RealtimePublisher
   void publishingLoop()
   {
     is_running_ = true;
-    turn_ = State::REALTIME;
+    turn_.store(State::NON_REALTIME, std::memory_order_release);
 
     while (keep_running_) {
       MessageT outgoing;
 
-      // Locks msg_ and copies it
-
-#ifdef NON_POLLING
-      std::unique_lock<std::mutex> lock_(msg_mutex_);
-#else
-      lock();
-#endif
-
-      while (turn_ != State::NON_REALTIME && keep_running_) {
-#ifdef NON_POLLING
-        updated_cond_.wait(lock_);
-#else
-        unlock();
-        std::this_thread::sleep_for(std::chrono::microseconds(500));
-        lock();
-#endif
+      {
+        // Locks msg_ and copies it to outgoing
+        std::unique_lock<std::mutex> lock_(msg_mutex_);
```

Comment:
Not a big deal, but now that we're not locking and unlocking anymore, const std::lock_guard should be sufficient instead of std::unique_lock

Reply:
Also, I ran a test pre- and post-PR, and I replicated the results from before on my software stack. That is, pre-PR, the ros2_control node was running at ~74% CPU usage and post-PR, it was running at 50% CPU usage.
Thanks again for the PR.

Reply:
unique_lock is being used for the conditional variable wait method

Reply:
Ah, sure. Missed that. Ignore my comment.